File: srm/srm-gateway/src/main/java/com/srm/gateway/entity/ContactPerson.java
package com.srm.gateway.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("contact_person")
public class ContactPerson {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long supplierId;
    
    private String name;
    
    private String position;
    
    private String phone;
    
    private String email;
    
    private String department;
    
    private Boolean isPrimary;
    
    private String remark;
    
    @TableField(fill = FieldFill.INSERT)
    private String createdBy;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedBy;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    @TableLogic
    private Integer delFlag;
}

File: srm-frontend-app/src/services/contactPerson.ts
import api from './api';
import type { ApiResponse } from '../types';

export interface ContactPerson {
  id: number;
  supplierId: number;
  name: string;
  position?: string;
  phone?: string;
  email?: string;
  department?: string;
  isPrimary?: boolean;
  remark?: string;
  createdAt?: string;
  updatedAt?: string;
  delFlag: number;
}

export interface ContactPersonCreate {
  supplierId: number;
  name: string;
  position?: string;
  phone?: string;
  email?: string;
  department?: string;
  isPrimary?: boolean;
  remark?: string;
}

export const contactPersonApi = {
  getBySupplierId: (supplierId: number): Promise<ApiResponse<ContactPerson[]>> => {
    return api.get(`/api/contact-persons/supplier/${supplierId}`);
  },

  getById: (id: number): Promise<ApiResponse<ContactPerson>> => {
    return api.get(`/api/contact-persons/${id}`);
  },

  create: (data: ContactPersonCreate): Promise<ApiResponse<number>> => {
    return api.post('/api/contact-persons', data);
  },

  update: (id: number, data: Partial<ContactPerson>): Promise<ApiResponse<void>> => {
    return api.put(`/api/contact-persons/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return api.delete(`/api/contact-persons/${id}`);
  },
};
